Cargo.toml

```diff
@@ -11,32 +11,32 @@ documentation = "https://docs.rs/secp256k1/"
 description = "Rust bindings for Pieter Wuille's `libsecp256k1` library. Implements ECDSA for the SECG elliptic curve group secp256k1 and related utilities."
 keywords = [ "crypto", "ECDSA", "secp256k1", "libsecp256k1", "bitcoin" ]
 readme = "README.md"
-build = "build.rs"
-links = "secp256k1"
 
 # Should make docs.rs show all functions, even those behind non-default features
 [package.metadata.docs.rs]
 features = [ "rand", "serde", "recovery", "endomorphism" ]
 
-[build-dependencies]
-cc = ">= 1.0.28, < 1.0.42"
-
 [lib]
 name = "secp256k1"
 path = "src/lib.rs"
 
 [features]
 unstable = []
 default = ["std"]
-std = ["rand/std"]
-recovery = []
-endomorphism = []
-lowmemory = []
+std = ["secp256k1-sys/std", "rand/std"]
+recovery = ["secp256k1-sys/recovery"]
+endomorphism = ["secp256k1-sys/endomorphism"]
+lowmemory = ["secp256k1-sys/lowmemory"]
+
+# Use this feature to not compile the bundled libsecp256k1 C symbols,
+# but use external ones.
+external-symbols = ["secp256k1-sys/external-symbols"]
```

Comment:
Can you just *not* depend on secp256k1-sys at all if you build with external-symbols? pulling in a dummy crate as a dependency sucks.

Reply:
The crate is still defining the C interface and the FFI objects used in it. In fact it's not doing less than when built without `external-symbols`, it's just the build.rs construction is no longer utilized.

Reply:
At least we should add a scarier warning.

Also, as you noted in another channel, this has the potential to cause UB a user is using an old version of rust-secp, as well as this one with `external-symbols` enabled...so we can't depend on cargo failing if this is enabled on a cargo build.